docs/man/rpm-dependency-generators.7.scd

```diff
@@ -0,0 +1,381 @@
+*%\_\_*_NAME_*\_enhances_opts* _BODY_++
+*%\_\_*_NAME_*\_obsoletes_opts* _BODY_++
+*%\_\_*_NAME_*\_orderwithrequires_opts* _BODY_++
+*%\_\_*_NAME_*\_provides_opts* _BODY_++
+*%\_\_*_NAME_*\_recommends_opts* _BODY_++
+*%\_\_*_NAME_*\_requires_opts* _BODY_++
+*%\_\_*_NAME_*\_suggests_opts* _BODY_++
+*%\_\_*_NAME_*\_supplements_opts* _BODY_
+
+*%\_\_requires_exclude* _REGEX_++
+*%\_\_requires_exclude_from* _REGEX_++
+*%\_\_provides_exclude* _REGEX_++
+*%\_\_provides_exclude_from* _REGEX_
+
+# DESCRIPTION
+
+RPM uses a pluggable interface coupled with a file classifier to add dependency
+information to packages at build time. This system relies on specialized, domain
+specific executables, called _generators_, that do the heavy-lifting of figuring
+out the dependencies from the files present in the buildroot. It complements the
+manual tags, such as *Requires* or *Provides*, in *rpm-spec*(5) files.
+
+Generators are paired with logical labels, called _file attributes_, assigned to
+the packaged files by the file classifier. This ensures that every file is only
+processed by the applicable generator(s), as opposed to all generators, which
+simplifies their implementation and significantly improves the overall build
+performance.
+
+# FILE ATTRIBUTES
+
+A file attribute is a logical label that RPM gives to one or more files in the
+buildroot, based on a set of matching _rules_. It can represent a single file
+type, such as ELF, or a broader domain, such as all source files written in a
+specific programming language.
+
+These attributes are distinct from the file attributes in the filesystem and are
+only kept in memory during package builds. A single file may have an arbitrary
+number of attributes, and a single attribute may be assigned to an arbitrary
+number of files.
+
+A valid file attribute called _NAME_ is introduced with an *rpm-macrofile*(5),
+_NAME.attr_, dropped into the *%{\_fileattrsdir}* directory, that defines at
+least one of the following macros:
+
+```
+%__NAME_path
+%__NAME_magic
+%__NAME_exclude_path
+%__NAME_exclude_magic
+```
+
+The values of these macros represent the inclusion and exclusion rules. They are
+macro-expanded on use and then interpreted as extended regular expressions (RE).
+Matching is inclusive unless changed by _flags_ (see below): If the path, magic
+or both match, the file is said to _have_ the attribute in question, unless
+there is a matching exclusion rule (also inclusive by default) or a flag that
+prevents the match.
+
+The path REs are matched against the packaged path, that is, without the
+*%buildroot* prefix (for example, _/bin/bash_). The magic REs are matched
+against the result of *magic*(5) classification and possibly the *FILECLASS* tag
+stored in the package (see *--fileclass* in *rpm*(8) for details).
+
+Flags are comma-separated lists specified with the following macros:
+
+```
+%__NAME_flags
+%__NAME_exclude_flags
+```
+
+Currently, the following flags are supported (Added: 4.9.1):
```

Comment:
BTW 4.9.1 is such ancient history by now, I think we can safely drop that Added bit. 

The oldest Added mention prior to this is on 4.12.0 and that already seems a bit over the top. Okay, RHEL 7 and its contemporaries are still out there, so maybe that's one way to draw that line (ie 4.11)

Reply:
We have the unwritten rule that rpm needs to be buildable by RHEL-latest, so another unwritten rule could be documentation needing to cover the oldest RHEL still alive :smile: Of course these being unwritten rules, there's always room for manouvering if eg something is important enough.